drivers/volume/portworx/portworx.go

```diff
@@ -1666,6 +1295,311 @@ func (p *portworx) UpdateMigratedPersistentVolumeSpec(
 	return object, nil
 }
 
+func (p *portworx) CreateGroupSnapshot(snap *stork_crd.GroupVolumeSnapshot) (
+	*storkvolume.GroupSnapshotCreateResponse, error) {
+	ok, msg, err := p.ensureNodesHaveMinVersion("2.0.2")
+	if err != nil {
+		return nil, err
+	}
+
+	if !ok {
+		err = &errors.ErrNotSupported{
+			Feature: "Group snapshots using CRD",
+			Reason:  "Only supported on PX version 2.0.2 onwards: " + msg,
+		}
+
+		return nil, err
+	}
+
+	volNames, err := k8sutils.GetVolumeNamesFromLabelSelector(snap.Namespace, snap.Spec.PVCSelector.MatchLabels)
+	if err != nil {
+		return nil, err
+	}
+
+	snapType, err := getSnapshotType(snap.Spec.Options)
+	if err != nil {
+		return nil, err
+	}
+	switch snapType {
+	case crdv1.PortworxSnapshotTypeCloud:
+		return p.createGroupCloudSnapFromVolumes(snap, volNames, snap.Spec.Options)
+	case crdv1.PortworxSnapshotTypeLocal:
+		return p.createGroupLocalSnapFromPVCs(snap, volNames, snap.Spec.Options)
+	default:
+		return nil, fmt.Errorf("unsupported snapshot type: %s", snapType)
+	}
+}
+
+func (p *portworx) GetGroupSnapshotStatus(snap *stork_crd.GroupVolumeSnapshot) (
+	*storkvolume.GroupSnapshotCreateResponse, error) {
+	snapType, err := getSnapshotType(snap.Spec.Options)
+	if err != nil {
+		return nil, err
+	}
+	switch snapType {
+	case crdv1.PortworxSnapshotTypeCloud:
+		return p.getGroupCloudSnapStatus(snap)
+	case crdv1.PortworxSnapshotTypeLocal:
+		return nil, fmt.Errorf("status API not supported for local group snapshots")
```

Comment:
This can't return an error. The caller wouldn't know how to deal with a legit error as opposed to this error.